features/netsocket/emac-drivers/TARGET_Freescale_EMAC/mbed_lib.json

```diff
@@ -1,7 +1,7 @@
 {
     "name": "kinetis-emac",
     "config": {
-        "rx-ring-len": 16,
-        "tx-ring-len": 8
+        "rx-ring-len": 2,
```

Comment:
Doing this without correspondingly reducing the lwIP mem config is a little "sneaky" - effectively you've giving this specific platform a huge memory pool for lwIP, because it was set big in a target override apparently to make room for these buffers.

I'd like you to either

a) reduce the target-override memory pool for lwIP by the corresponding amount - if you can (but I assume you do need the extra space, so you can't).
b) reduce by less, and raise the default memory pool (and other target overrides?) by the difference - so that all platforms get the extra memory you've found you need. Maybe the other targets with big pools can also have their buffer count reduced, so they don't need to get bigger.

Reply:
I'll lower the `lwip.mem-size` value for some amount..
This 16, was actually causing `memory_manager->alloc_heap(ENET_ETH_MAX_FLEN, ENET_BUFF_ALIGNMENT);` called 16 times.. Where `ENET_ETH_MAX_FLEN` is 1522 bytes and alignment is 16. I'm unsure of how much the allocator overhead is.

Total memory usage is something between 24352 bytes 24608 bytes.

As a result, I lowered the `lwip.mem-size` for Freescale boards from 36560 to 16384 (~20 kB less), which still seems to pass the Cloud Client FW update test.

Reply:
If 16K is required with 2 RX buffers, that does suggest cloud wants about 12K of lwIP mem workspace, so maybe that should be the default mem-size? (I believe we assume that target defaults should be good for cloud client). And then other targets should be doing 12K + their buffer size?

I guess the issue is that any upward increase like that might not be appropriate for a patch release, so should be done separately. Something to follow up.